cmd/gcp-controller-manager/kubelet_readonly_csr_approver_test.go

```diff
@@ -169,22 +169,20 @@ func TestKubeletReadonlyApprover(t *testing.T) {
 			approver := newKubeletReadonlyCSRApprover(request.controllerContext)
 			*autopilotEnabled = testCase.autopilotEnabled
 			err := approver.handle(request.context, request.csr)
-			if testCase.expectError != nil && err == nil {
-				t.Fatalf("expect no error but gor: %v", err)
-			}
-			if err != nil {
-				if testCase.expectError == nil {
-					t.Fatalf("error when handle approver, error: %v", err)
-				}
-				if err.Error() != testCase.expectError.Error() {
-					t.Fatalf("error not match, expect: %v, got: %v", testCase.expectError, err)
-				}
-			} else {
+			switch {
+			case testCase.expectError == nil && err == nil:
 				if diff := cmp.Diff(request.csr.Status.Conditions[0], testCase.expectCondition, cmp.Comparer(func(x, y capi.CertificateSigningRequestCondition) bool {
 					return x.Type == y.Type && x.Reason == y.Reason && x.Message == y.Message && x.Status == y.Status
 				})); diff != "" {
 					t.Fatalf("condition don't match, diff -want +got\n%s", diff)
 				}
+				break
+			case testCase.expectError != nil && err == nil:
+				t.Fatalf("expect no error but gor: %v", err)
+			case err != nil && testCase.expectError == nil:
+				t.Fatalf("error when handle approver, error: %v", err)
+			case err.Error() != testCase.expectError.Error():
+				t.Fatalf("error not match, expect: %v, got: %v", testCase.expectError, err)
```

Comment:
these can be combined, no?

case testCase.expectedError != nil && err == nil,
         err != nil && testCase.expectedError == nil,
         err.Error() != testCase.expectedError():
           t.Fatalf("got error: %v; want error: %v", err, testCase.expectError)

Reply:
Done